docs/design/enqueue-action.md

```diff
@@ -0,0 +1,59 @@
+# Enqueue
+
+## Introduction
+
+In Volcano there are 6 actions such as `enqueue`, `allocate`, `preempt`, `reclaim`, `backfill`, `shuffle` and with the help of
+plugins like `conformance`, `drf`, `gang`, `nodeorder` and more plugins. All these plugins provides
+behavioural characteristics how scheduler make scheduling decisions.
+
+## Enqueue Action
+
+Enqueue is one of the actions in Volcano scheduler.  Enqueue action comes into play when
+a new job is created, this action will filter qualified jobs into the queue to be scheduled.
+
+When the minimum number of resource requests under a job can not be met, even if the scheduling action is 
+performed for pod under a Job, pod will not be schedule because the "Gang" constraint is not reached. 
+state of job only allowed change from `Pending` to `Inqueue` if the minimum resource size of the job is met.
+
+Enqueue action is the preparatory stage in the scheduling process. it can prevent a large number of 
+unscheduled pods in the cluster and improve the performance of the scheduler in the high-load scenarios where 
+the cluster resources may be insufficient. Therefore, the Enqueue action is an essential action for the scheduler configuration.
+
+In Enqueue Action, there are multiple plugin functions that are getting used like
+
+1. QueueOrderFn(Plugin: Priority, DRF, Proportion),
+2. JobOrderFn(Plugin: Priority, DRF, Gang),
+3. JobEnqueueable(Plugin: OverCommit, Proportion, ResourceQuota, SLA),
```

Comment:
4. JobEnqueued(Plugin: OverCommit)

This interface was used to update resource infos after one job turned `inqueue`, `overcommit` plugin need this interface to update `inqueue` resources in each queue and whole cluster.